Cabal/Distribution/Simple/Haddock.hs

```diff
@@ -209,24 +210,29 @@ haddock pkg_descr lbi suffixes flags' = do
            warn (fromFlag $ haddockVerbosity flags)
              "Unsupported component, skipping..."
            return ()
+        -- Don't print this message if nothing is going to happen...
```

Comment:
It's saying that we shouldn't print this message if no Haddocking will occur. Previously, the setup message was unconditionally printed. Now we put the action in smsg and call it inside the cases.